src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalyzer.java

```diff
@@ -19,181 +19,121 @@
-import java.util.HashSet;
-import java.util.Set;
-
-import org.sonarlint.intellij.issue.IssueProcessor;
-import org.sonarlint.intellij.messages.TaskListener;
-import org.sonarlint.intellij.trigger.TriggerType;
+import com.intellij.openapi.vfs.encoding.EncodingProjectManager;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.sonarlint.intellij.core.SonarLintFacade;
+import org.sonarlint.intellij.core.SonarLintServerManager;
 import org.sonarlint.intellij.ui.SonarLintConsole;
 import org.sonarlint.intellij.util.SonarLintUtils;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.IssueListener;
 
-public class SonarLintAnalyzer extends AbstractProjectComponent {
-  private static final Logger LOGGER = Logger.getInstance(SonarLintAnalyzer.class);
-  private final IssueProcessor processor;
-  private final MessageBus messageBus;
-  private final JobQueue queue;
-  // used to synchronize the handling of queue and running status together
-  private final Object lock;
-  private final SonarLintStatus status;
-  private final SonarLintConsole console;
-
-  public SonarLintAnalyzer(Project project, IssueProcessor processor) {
-    super(project);
-    this.processor = processor;
-    this.messageBus = project.getMessageBus();
-    this.queue = new JobQueue(project);
-    this.lock = new Object();
-    this.status = SonarLintStatus.get(this.myProject);
-    this.console = SonarLintConsole.get(myProject);
-
-    messageBus.connect(project).subscribe(TaskListener.SONARLINT_TASK_TOPIC, new TaskListener() {
-      @Override public void started(SonarLintJob job) {
-        //nothing to do
-      }
-
-      @Override public void ended(SonarLintJob job) {
-        taskFinished();
-      }
-    });
-  }
-
-  public void submitAsync(Module m, Collection<VirtualFile> files, TriggerType trigger) {
-    if (console.debugEnabled()) {
-      SonarLintConsole.get(myProject).debug(String.format("[%s] %d file(s) submitted", trigger.getName(), files.size()));
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class SonarLintAnalyzer {
+  public AnalysisResults analyzeModule(Module module, Collection<VirtualFile> filesToAnalyze, IssueListener listener) {
+    Project p = module.getProject();
+    SonarLintConsole console = SonarLintConsole.get(p);
+    SonarLintServerManager core = SonarLintUtils.get(SonarLintServerManager.class);
+
+    // Configure plugin properties. Nothing might be done if there is no configurator available for the extensions loaded in runtime.
+    Map<String, String> pluginProps;
+    AnalysisConfigurator analysisConfigurator = module.getComponent(AnalysisConfigurator.class);
```

Comment:
Currently yes, because there is only one for Java.
But to avoid problems in the future if we implement more, I will get the list. Good catch.